bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/checksum/DigestManager.java

```diff
@@ -158,6 +161,18 @@ private void verifyDigest(long entryId, ByteBuf dataReceived) throws BKDigestMat
         verifyDigest(entryId, dataReceived, false);
     }
 
+    private static final FastThreadLocal<ByteBuf> DIGEST_BUFFER = new FastThreadLocal<ByteBuf>() {
+        @Override
+        protected ByteBuf initialValue() throws Exception {
+            return PooledByteBufAllocator.DEFAULT.directBuffer(1024);
```

Comment:
The required ByteBuf size for each digest manager:
CRC32CDigestManager -> 4
CRC32DigestManager -> 8
DummyDigestManager -> 0
MacDigestManager -> 20
Do we need to allocate 1024 bytes?

Reply:
Just in case. Though this is only used for MAC at this point, and one per thread. It shouldn't be any noticeable waste.

Reply:
I actually was referring to my next change in the DigestManager :)